### PR TITLE
[bugfix] TF-2426 Insert link dialog is obscured

### PR DESCRIPTION
**Pull request by hieutbui**
## Issue
#2426 

## Relevant
https://github.com/linagora/html-editor-enhanced/pull/26

## Resolved

https://github.com/linagora/tmail-flutter/assets/80142234/8ba4fc80-9e90-48a0-902f-10e0139b6e55

### CODE REVIEW

File: lib/features/composer/presentation/composer_controller.dart [hunk 1]
```diff
@@ -2179,4 +2181,61 @@ class ComposerController extends BaseController {
...
+    bool? isOpenNewTab,
+    String linkTagId
+  ) async {
+    final displayTextController = TextEditingController(text: text);
+    final urlController = TextEditingController(text: url);
+    final isOpenNewTabValue = RxBool(isOpenNewTab ?? true);
+    final GlobalKey<FormState> insertLinkDialogKey = GlobalKey<FormState>();
+    final displayTextFieldFocusNode = FocusNode();
+    final urlFieldFocusNode = FocusNode();
+    final isOpenNewTabFocusNode = FocusNode();
```
Comment (dab246) [inline lib/features/composer/presentation/composer_controller.dart:2198]: Please dispose it when dialog disposed

Comment (hieutbui) [reply lib/features/composer/presentation/composer_controller.dart]: done

File: lib/features/composer/presentation/composer_controller.dart [hunk 2]
```diff
@@ -139,6 +139,8 @@ class ComposerController extends BaseController {
   final ccEmailAddressController = TextEditingController();
   final bccEmailAddressController = TextEditingController();
   final searchIdentitiesInputController = TextEditingController();
+  final displayTextInsertLinkDialogController = TextEditingController();
+  final urlInsertLinkDialogController = TextEditingController();
```
Comment (dab246) [inline lib/features/composer/presentation/composer_controller.dart:143]: Why not include it in the dialog widget? It should be managed by dialog

File: lib/features/composer/presentation/composer_controller.dart [hunk 3]
```diff
@@ -153,6 +155,10 @@ class ComposerController extends BaseController {
   FocusNode? bccAddressFocusNode;
   FocusNode? searchIdentitiesFocusNode;
 
+  final displayTextFieldInsertLinkDialogFocusNode = FocusNode();
+  final urlFieldInsertLinkDialogFocusNode = FocusNode();
+  final isOpenNewTabInsertLinkDialogFocusNode = FocusNode();
```
Comment (dab246) [inline lib/features/composer/presentation/composer_controller.dart:160]: idem

File: lib/features/composer/presentation/composer_controller.dart (hunk 2 above)
Comment (hieutbui) [reply lib/features/composer/presentation/composer_controller.dart]: done

File: lib/features/composer/presentation/composer_controller.dart (hunk 3 above)
Comment (hieutbui) [reply lib/features/composer/presentation/composer_controller.dart]: done

**Comment by dab246** [comment]
@hieutbui  Please rebase it

**Review by dab246 — APPROVED** [review]
LGTM

**Pull request merged**